omnipaxos/src/messages.rs

```diff
@@ -73,16 +87,11 @@ pub mod sequence_paxos {
         pub n: Ballot,
         /// The sequence number of this message in the leader-to-follower accept sequence
         pub seq_num: SequenceNumber,
-        /// The decided snapshot.
-        pub decided_snapshot: Option<SnapshotType<T>>,
-        /// The log suffix.
-        pub suffix: Vec<T>,
-        /// The index of the log where the entries from `suffix` should be applied at (also the compacted idx of `decided_snapshot` if it exists)
-        pub sync_idx: u64,
         /// The decided index
-        pub decided_idx: u64,
-        /// StopSign to be accepted
-        pub stopsign: Option<StopSign>,
+        pub decided_idx: usize,
```

Comment:
When we use snapshots the sync_idx will equal the decided_idx the majority of the time. The exception to this is when a stopsign is decided, then sync_idx = decided_idx - 1.

Reply:
Not 100% sure what you mean about the Decide message but the follower will adopt the decided index and undecided suffix  of the leader from the AcceptSync, and then reply with an Accepted which the leader will eventually use to see that the undecided entries have become chosen and reply appropriately.

Reply:
let's say we have decided up to 100 entries in round 2. The follower only had 40 entries from round 1. Then it should get the suffix of 60 entries that it should apply at index 40 (this used to be called the sync_idx) and the decided _idx should be 100. 

But from your comment it is as I suspected, it will send the decided_idx (i.e., 100) here in a subsequent message.